lib/xandra/protocol.ex

```diff
@@ -538,6 +554,30 @@ defmodule Xandra.Protocol do
   defp new_page(%Prepared{result_columns: result_columns}),
     do: %Page{columns: result_columns}
 
+  defp enhance_column_types(columns, options) do
```

Comment:
I think we could call this one something like `use_correct_type_for_dates` for now, until we have something else that justify the genericness of `enhance_column_types`. Wdyt?

Reply:
I really like `enhance_column_types` more than `use_correct_type_for_dates`, especially since there is no such thing as "correct type" IMO. :bowtie:

Reply:
Yeah but "enahnce_column_types" really doesn't say what this is doing and we are not strictly *enhancing* the column type. I think we need a better name.

Reply:
`rewrite_column_types`?

Reply:
Yep that sounds good.